src/track/beatutils.cpp

```diff
@@ -29,20 +31,224 @@ const int kHistogramDecimalPlaces = 2;
+        // value so we don't consider changes to a neighboor value in the ordered tempoFrequency table
+        if (newStableTempo == stableTemposByPosition.last()) {
+            continue;
+        // Here we check if the new tempo is the right neighboor of the previous tempo
+        } else if (stableTemposByPosition.last() != tempoFrequency.lastKey() and
+                newStableTempo == (tempoFrequency.find(stableTemposByPosition.last()) + 1).key()) {
+            continue;
+        // Here we check if the new tempo is the left neighboor of the previous tempo
+        } else if (stableTemposByPosition.last() != tempoFrequency.firstKey() and
+                newStableTempo == (tempoFrequency.find(stableTemposByPosition.last()) - 1).key()) {
+            continue;
+        } else {
+            // This may not be case when our median window is even, we can't use it
+            // Because find will return an iterator pointing to end that we will *
+            if (tempoFrequency.contains(newStableTempo)) {
+                lastBeatChange = currentBeat - filterTempo.lag() - stabilzeTempo.lag();
+                stableTemposByPosition[lastBeatChange] = newStableTempo;
+            }
+        }
+    }
+    stableTemposByPosition[tempoList.count()] = median;
+    return stableTemposByPosition;
+}
+
+void BeatUtils::RemoveSmallArrhythmic(
+        QVector<double>& rawBeats,  int sampleRate, QMap<int, double> &stableTemposByPosition) {
+    // A common problem the analyzer has is to detect arrhythmic regions
+    // of tacks with a constant tempo as in a different unsteady tempo. 
+    // This happens frequently on builds and breaks with heavy effects on edm music.
+    // Since these occurs most on beatless regions we do not want them to be
+    // on a different tempo, because they are still syncable in the true tempo
+    // We arbitraly remove these arrhythmic regions if they are short than 16s
+    auto positionsWithTempoChange = stableTemposByPosition.keys();
+    auto tempoValues = stableTemposByPosition.values();
+    QVector<double> fixedBeats;
+    fixedBeats.reserve(rawBeats.size());
+    fixedBeats << QVector<double>::fromStdVector(std::vector<double>(
+                rawBeats.begin(), rawBeats.begin() + positionsWithTempoChange[1]));
+
+    for (int i = 2; i < positionsWithTempoChange.size(); i+=1) {
+        int smallInBeats =  16 / (60 / tempoValues[i-1]);
+        int limitAtLeft = positionsWithTempoChange[i-1];
+        int limitAtRight = positionsWithTempoChange[i];
+        int lenghtOfChange = limitAtRight - limitAtLeft;
+        double previousTempo = (stableTemposByPosition.lowerBound(limitAtLeft)-1).value();
+
+        if (lenghtOfChange <= smallInBeats) {
+            stableTemposByPosition.remove(limitAtLeft);
+            double beatOffset = rawBeats[limitAtLeft];
+            double beatLength = floor(((60.0 * sampleRate) / previousTempo) + 0.5);
```

Comment:
what is the rounding for?

Reply:
I think that needs to be removed.

If we add beats here, the later ironing will treat them as strong reference points and will apply the 0,25 ms rule to it. But that is not necessary.  We can just advance to the next beat and put them later exactly on the calculated average.

Reply:
hm, just removing the beats was my first idea but that does not work without an additional structure to keep track that we need to add the beats in this region later, and the big jump will result in very large beat length that will completely mess the average so removing them adds a lot of bookkeeping.
Maybe we can set the beats to -1 or something like that and use that idea of beats not increasing as a way to tell not to consider these beats on iron code and just add the avarage length. I will try that, should work...